asmcomp/schedgen.ml

```diff
@@ -44,18 +44,21 @@ let dummy_node =
    - code_checkbounds contains the latest checkbound node not matched
      by a subsequent load or store. *)
 
-let code_results = (Hashtbl.create 31 : (location, code_dag_node) Hashtbl.t)
-let code_uses = (Hashtbl.create 31 : (location, code_dag_node) Hashtbl.t)
-let code_stores = ref ([] : code_dag_node list)
-let code_loads = ref ([] : code_dag_node list)
-let code_checkbounds = ref ([] : code_dag_node list)
-
-let clear_code_dag () =
-  Hashtbl.clear code_results;
-  Hashtbl.clear code_uses;
-  code_stores := [];
-  code_loads := [];
-  code_checkbounds := []
+type code_dag =
+  { results : (location, code_dag_node) Hashtbl.t
+  ; uses : (location, code_dag_node) Hashtbl.t
+  ; mutable stores : code_dag_node list
+  ; mutable loads : code_dag_node list
+  ; mutable checkbounds : code_dag_node list
+  }
+
+let create () =
```

Comment:
I find `create ()` unclear at the call site (another way to think about this: what would Schedgen.create do? It is unclear, so the name is not great), what about `let empty_dag ()` for example?